xontrib/prompt_ret_code.xsh

```diff
@@ -2,20 +2,31 @@ from xonsh.tools import ON_WINDOWS as _ON_WINDOWS
 
 
 def _ret_code_color():
+    try:
+	color_bg = $PROMPT_RET_CODE_BACKGROUND
+    except:
+	color_bg = False
+
+    if color_bg:
+	prefix = 'BACKGROUND'
```

Comment:
Another option here is to allow the env-var to take the values of `BOLD` or `BACKGROUND` (and potentially others) and then use the value directly.

Take a look in `xonsh.environ` at the `DEFAULT_ENSURERS` for how we type and validate env-var inputs

You can create an ensurer for the acceptable value types and then add that ensurer using

```
${...}.set_ensurer(env_var_name, ensurer_func)
```

Reply:
Good idea. My proposed changes extend hard coded colors to slightly more hard coded colors, but the functionality should probably be flexible enough to have any color string that would be valid in `$PROMPT_FIELDS`. This would only require a variable for each of zero or non-zero return codes.

I'll have a go setting up an ensurer for these two variables to make sure they are valid color strings.

Reply:
Actually - it would be 4 variables. zero and non-zero for background and foreground.